test/fixtures/dialects/tsql/triggers.yml

```diff
@@ -33,7 +33,7 @@ file:
             - comma: ','
             - literal: '10'
             - end_bracket: )
-    statement_terminator: ;
+      - statement_terminator: ;
 - go_statement:
     keyword: GO
 - batch:
```

Comment:
This is inconsistent behaviour with the `ansi` dialect. I would expect the terminator to be outside the `statement` segment.

Reply:
I did look into these (should have commented).

It's due to this slight change included in here:

<img width="686" alt="image" src="https://user-images.githubusercontent.com/10931297/161319172-7a300d73-bdb5-4162-981c-e8009d60f72d.png">

The original definition didn't allow terminators, but [looking at the grammar](https://docs.microsoft.com/en-us/sql/t-sql/statements/create-trigger-transact-sql?view=sql-server-ver15) this was incorrect and it should have.

And because TSQL is odd and the terminator is optional (in both the substatement and the main statement) that leads to these sorts of this. 

It could be parsed as this:

```sql
CREATE TRIGGER example
AS
    SELECT * FROM table;
```

Or this:

```sql
CREATE TRIGGER example
AS
    SELECT * FROM table
;
```

Both are equally valid.

The other statements have this same issue (and have always had), but you see the change here because I've corrected the grammar.

So I don't think it's an issue. That optional semi-colon in tsql is just weird!

Reply:
Interesting 🤔 . I'm going to check out this branch and have a play with it.

Reply:
Ok I see why this a little more complicated. There are two nested statements, so it's easy to see why the terminator could be seen to terminate the inner one or the outer one. My reading of the query however is that the terminator, terminated the `CREATE` query, not just the `RaiseError` statement. Given that, I think the original placement is where it _should_ be.

Reply:
Don't disagree, but as @WittierDinosaur said, Delimted is currently greedy (is this fixed in your refactoring?) so currently it's put in the other place. However as it's optional in both it doesn't really matter one way of the other. And as I say, other TSQL statements already have the same issue.

So don't think this needs to block this PR. I can revert that change if you're concerned but technically the previous grammar was incorrect, even if it parsed these examples "better".

Reply:
Ok - I think I've got a solution which I think is a better representation of the underlying structure. I'm going to push to the same branch in the interests of ease, but @tunetheweb feel free to keep editing obv.